src/main/java/com/genesys/statistics/Notifications.java

```diff
@@ -0,0 +1,110 @@
+            
+            listeners.entrySet().forEach((entry) -> {
+                final String name = entry.getKey();
+                final Collection<NotificationListener> list = entry.getValue();
+
+                client.getChannel(name).subscribe((channel, message) ->  {
+                    Map<String, Object> data = message.getDataAsMap();
+                    list.forEach(listener -> listener.onNotification(name, data));
+                }, (channel, message) -> {
+                    String subscription = (String) message.get("subscription");
+                    if(message.isSuccessful()) {
+                        logger.debug("Successfuly subscribed to channel: {}", subscription);
+                    }
+                    else {
+                        logger.error("Cannot subscribe to channel: {}", subscription);
+                    }
+                });
+            });
+        }
+        else {
+            logger.error("Cometd handshake failed");
+            logger.debug(msg.toString());
+        }
+    }
+    
+    public void initialize(String endpoint, String apiKey, String token) throws StatisticsException {
+        try {
+            httpClient = new HttpClient(new SslContextFactory());
+            httpClient.start();
+            client = new BayeuxClient(endpoint, new LongPollingTransport(new HashMap<String, Object>(), httpClient) {
+                final TraceInterceptor interceptor = new TraceInterceptor();
+                
+                @Override
+                protected void customize(Request request) {
+                    request.header("x-api-key", apiKey);
+                    request.header("Authorization", String.format("Bearer %s", token));
+                    request.header(TraceInterceptor.TRACEID_HEADER, interceptor.makeUniqueId());
+                    request.header(TraceInterceptor.SPANID_HEADER, interceptor.makeUniqueId());
+
+                    logger.debug("{}\n{}", request.toString(), request.getHeaders().toString());
+                }                
+            });
+            
+            initialize(client);
+        }
+        catch(Exception ex) {
+            throw new StatisticsException("Cometd initialization failed.", ex);
+        }
+    }
+    
+    void initialize(BayeuxClient client) {
+        this.client = client;
+        
+        logger.debug("Starting cometd handshake...");
+        client.handshake((channel, msg) -> onHandshake(msg));
+    }
+    
+    public void disconnect() throws StatisticsException  {
+        try {
+            if(client != null) {
+                client.disconnect();
+            }
+            if(httpClient != null) {
+                httpClient.stop();
+            }
+        }
+        catch (Exception ex) {
+            throw new StatisticsException("Cannot disconnection", ex);
+        }
+    }
+    
+    public void subscribe(String channelName, NotificationListener listener) {
```

Comment:
it can be simplified to:
listeners.computeIfAbsent(channelName, k -> new ConcurrentLinkedQueue<>()).add(listener);